### ISSUE TITLE
[2.1.0~alpha] opam switch create -y . fails

### ISSUE THREAD

**Issue by bikallem** [ocaml/opam#4169]
I am trying to install the dune ```dev-switch``` with the new opam v2.1.0~alpha. However, when I do 
```opam switch create -y . --deps-only --with-test --debug-level=3```, it fails with the following error/output.

```
00:00.004  GSTATE                 LOAD-GLOBAL-STATE @ /home/user1/.opam
00:00.004  SYSTEM                 LOCK /home/user1/.opam/lock (none => read)
00:00.004  SYSTEM                 LOCK /home/user1/.opam/config.lock (none => write)
00:00.006  FILE(config)           Read ~/.opam/config in 0.002s
00:00.006  RSTATE                 LOAD-REPOSITORY-STATE @ /home/user1/.opam
00:00.006  FILE(repos-config)     Read ~/.opam/repo/repos-config in 0.000s
00:00.006  SYSTEM                 LOCK /home/user1/.opam/repo/state.cache (none => read)
00:00.047  RSTATE                 Loaded /home/user1/.opam/repo/state.cache in 0.042s
00:00.075  SYSTEM                 LOCK /home/user1/.opam/repo/state.cache (read => none)
00:00.075  RSTATE                 Cache found
00:00.075  SWACT                  create_empty_switch at /home/user1/github/dune
00:00.075  SYSTEM                 mkdir /home/user1/github/dune/_opam
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/lib
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/lib/stublibs
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/lib/toplevel
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/.opam-switch
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/.opam-switch/build
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/bin
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/sbin
00:00.076  SYSTEM                 mkdir /home/user1/github/dune/_opam/doc
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/.opam-switch/install
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/.opam-switch/config
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man1
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man1M
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man2
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man3
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man4
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man5
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man6
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man7
00:00.077  SYSTEM                 mkdir /home/user1/github/dune/_opam/man/man9
00:00.077  SWACT                  install_switch_config switch=/home/user1/github/dune
00:00.077  FILE(switch-config)    Wrote /home/user1/github/dune/_opam/.opam-switch/switch-config in 0.000s
00:00.078  FILE(config)           Wrote /home/user1/.opam/config in 0.000s
00:00.078  STATE                  LOAD-SWITCH-STATE @ /home/user1/github/dune
00:00.078  FILE(switch-config)    Read ~/github/dune/_opam/.opam-switch/switch-config in 0.000s
00:00.078  FILE(switch-config)    Read ~/github/dune/_opam/.opam-switch/switch-config in 0.000s
00:00.078  FILE(switch-config)    Read ~/github/jsoo_todomvc/_opam/.opam-switch/switch-config in 0.000s
00:00.078  SYSTEM                 LOCK /home/user1/github/dune/_opam/.opam-switch/lock (none => write)
00:00.078  FILE(switch-config)    Read ~/github/dune/_opam/.opam-switch/switch-config in 0.000s
00:00.078  FILE(switch-state)     Cannot find /home/user1/github/dune/_opam/.opam-switch/switch-state
00:00.110  FILE(switch-config)    Wrote /home/user1/github/dune/_opam/.opam-switch/switch-config in 0.000s
00:00.110  STATE                  Switch state loaded in 0.032s
00:00.145  AUXCMD                 autopin: { dune => git+file:///home/user1/github/dune#master, dune-private-libs => git+file:///home/user1/github/dune#master, dune-glob => git+file:///home/user1/github/dune#master, dune-configurator => git+file:///home/user1/github/dune#master, dune-build-info => git+file:///home/user1/github/dune#master, dune-action-plugin => git+file:///home/user1/github/dune#master }
00:00.145  UPDATE                 update-dev-packages
00:00.145  PARALLEL               Iterate over 0 task(s) with 3 process(es)
00:00.145  SWACT                  add-to-reinstall unpinned_only:false packages:{}
00:00.145  FILE(package-version-list)  Cannot find /home/user1/github/dune/_opam/.opam-switch/reinstall
00:00.146  FILE(opam)             Read ~/github/dune/dune.opam in 0.000s
00:00.153  FILE(environment)      Read ~/.opam/default/.opam-switch/environment in 0.000s
00:00.153  GSTATE                 Failed to evaluate global variable sys-ocaml-version: "ocamlc": command not found.
00:00.187  FILTER                 ERR: value_bool: "osx"
00:02.125  STATE                  depexts loaded in 1.910s
00:02.133  FILE(opam)             Read ~/github/dune/dune-private-libs.opam in 0.000s
00:02.133  FILE(opam)             Read ~/github/dune/dune-glob.opam in 0.000s
00:02.133  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
00:02.133  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
00:02.133  FILE(opam)             Read ~/github/dune/dune-configurator.opam in 0.000s
00:02.134  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
00:02.134  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
00:02.134  FILE(opam)             Read ~/github/dune/dune-build-info.opam in 0.000s
00:02.134  FILE(opam)             Read ~/github/dune/dune-action-plugin.opam in 0.000s
00:02.134  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
00:02.134  FILTER                 Warn: ignoring version constraint version: Undefined string filter value: version
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] y
00:02.138  FILE(config)           Wrote /home/user1/.opam/config in 0.000s
00:02.138  SYSTEM                 rmdir /home/user1/github/dune/_opam
00:02.141  SYSTEM                 LOCK /home/user1/github/dune/_opam/.opam-switch/lock (write => none)
00:02.141  SYSTEM                 LOCK /home/user1/.opam/config.lock (write => none)
00:02.141  SYSTEM                 LOCK /home/user1/.opam/repo/lock (none => none)
00:02.141  SYSTEM                 LOCK /home/user1/.opam/config.lock (none => none)
Fatal error:
Not_found

```
opam config report:
```
# opam config report
# opam-version         2.1.0~alpha 
# self-upgrade         no
# system               arch=x86_64 os=linux os-distribution=rhel os-version=8.2
# solver               builtin-mccs+glpk
# install-criteria     -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria     -removed,-count[version-lag,solution],-new
# jobs                 3
# repositories         1 (http) (default repo at 88a179f5)
# pinned               1 (git)
# current-switch       default
# ocaml:native         true
# ocaml:native-tools   true
# ocaml:native-dynlink true
# ocaml:stubsdir       /home/user1/.opam/default/lib/ocaml/stublibs:/home/user1/.opam/default/lib/ocaml
# ocaml:preinstalled   false
# ocaml:compiler       4.10.0
```

**Comment by bikallem** [comment]
For more context, I am attempting to follow the dune Hacking.doc here - https://github.com/bikallem/dune/blob/master/HACKING.md

**Comment by rjbou** [comment]
This issue fix is in #4151
Thanks for the report!
